enhancements/coreos-bootimages.md

```diff
@@ -0,0 +1,143 @@
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Since the initial release of OpenShift 4, we have "pinned" RHCOS bootimage metadata inside [openshift/installer](https://github.com/openshift/installer).
+In combination with the binding between the installer and release image, this means that everything needed to install OpenShift (including the operating system "bootimages" such as e.g. AMIs and OpenStack `.qcow2` files) are all captured behind the release image which we can test and ship as an atomic unit.
+
+We have a mechanism to do [in place updates](https://github.com/openshift/machine-config-operator/blob/master/docs/OSUpgrades.md), but there is no automated mechanism to update "bootimages" past a cluster installation.
+
+This enhancement does not describe an automated mechanism to do this: the initial goal is to include this metadata in a standardized format in the cluster and at mirror.openshift.com so that UPI installations can do this manually, and we can start work on an IPI mechanism.
+
+### Stream metadata format
+
+As part of unifying Fedora CoreOS and RHEL CoreOS, we have standardized on the "stream metadata" format used by FCOS.  More in [FCOS docs](https://docs.fedoraproject.org/en-US/fedora-coreos/getting-started/) and [this RHCOS issue](https://github.com/openshift/os/issues/477).
+
+There is a new [stream-metadata-go](https://github.com/coreos/stream-metadata-go) library to consume this data.
+
+### Adding a new git repository and release image component + configmap with this data
+
+A new git repository https://github.com/openshift/rhel-coreos-bootimages will be created (based on [an existing prototype repository](https://github.com/cgwalters/rhel-coreos-bootimages)).  It will have the stream JSON.
+
+Additionally, this repository will be included in the release image as `rhel-coreos-bootimages` and it will use the CVO to install a `configmap/coreos-bootimages` in the `openshift-machine-config-operator` namespace.
+
+### Update `oc adm release new` to patch the installer binary to inject this data
+
+The installer needs CoreOS to start at all; today the bootimages are used for the bootstrap node
+and control plane which are created by the installer directly.  In order to avoid duplicating the
+metadata, change `oc adm release new` to *binary patch* the included `openshift-install`
+binary, in the same way we [do for the release image](https://github.com/openshift/installer/pull/1422).
+
+The RHCOS team will take over maintenance, CI, and automation of `rhel-coreos-bootimages`.  For OKD,
```

Comment:
> The RHCOS team will take over maintenance, CI, and automation of `rhel-coreos-bootimages`.

What does this entail?

Reply:
To start, we would do PRs to that repository to bump it in the same way we do openshift/installer.  See the prototype at https://github.com/cgwalters/rhel-coreos-bootimages

But we could switch to having a bot do PRs after new cosa builds happen, or something else.